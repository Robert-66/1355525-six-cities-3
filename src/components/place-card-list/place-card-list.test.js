import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCardList from './place-card-list';

const offers = [
  {
    id: 11,
    name: `Apartments Prinsengracht`,
    price: 120,
  },
  {
    id: 21,
    name: `Lovely Studio With Canal Views`,
    price: 99,
  },
  {
    id: 31,
    name: `Zandberg - Canal view apartments`,
    price: 140,
  },
  {
    id: 41,
    name: `Bright & new apartment with canal view`,
    price: 125,
  }
];

it(`PlaceCardList is rendered correctly`, () => {
  const placeCardList = renderer.create(
      <PlaceCardList
        offers={offers}
        onClickCardName={() => {}}
      />
  ).toJSON();

  expect(placeCardList).toMatchSnapshot();
});
