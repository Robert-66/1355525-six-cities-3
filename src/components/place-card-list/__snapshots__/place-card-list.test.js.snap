// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCardList is rendered correctly 1`] = `
<div
  className="places__list"
>
  <article
    className="place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
           
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="button place-card__bookmark-button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="/offer/11"
          onClick={[Function]}
        >
          Apartments Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
         
        Apartment
      </p>
    </div>
  </article>
  <article
    className="place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            99
          </b>
           
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="button place-card__bookmark-button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "62%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="/offer/21"
          onClick={[Function]}
        >
          Lovely Studio With Canal Views
        </a>
      </h2>
      <p
        className="place-card__type"
      >
         
        Apartment
      </p>
    </div>
  </article>
  <article
    className="place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            140
          </b>
           
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="button place-card__bookmark-button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="/offer/31"
          onClick={[Function]}
        >
          Zandberg - Canal view apartments
        </a>
      </h2>
      <p
        className="place-card__type"
      >
         
        Apartment
      </p>
    </div>
  </article>
  <article
    className="place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            125
          </b>
           
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="button place-card__bookmark-button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "98.00000000000001%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="/offer/41"
          onClick={[Function]}
        >
          Bright & new apartment with canal view
        </a>
      </h2>
      <p
        className="place-card__type"
      >
         
        Private Room
      </p>
    </div>
  </article>
  <article
    className="place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            125
          </b>
           
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="button place-card__bookmark-button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "70%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="/offer/51"
          onClick={[Function]}
        >
          Tile House
        </a>
      </h2>
      <p
        className="place-card__type"
      >
         
        Private Room
      </p>
    </div>
  </article>
  <article
    className="place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            210
          </b>
           
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="button place-card__bookmark-button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="/offer/61"
          onClick={[Function]}
        >
          Waterfront with extraordinary view
        </a>
      </h2>
      <p
        className="place-card__type"
      >
         
        Apartment
      </p>
    </div>
  </article>
</div>
`;
